b2sdk/transfer/inbound/downloader/parallel.py

```diff
@@ -328,14 +332,14 @@ def download_first_part(
         response.close()
 
         url = response.request.url
-        tries_left = 5 - 1  # this is hardcoded because we are going to replace the entire retry interface soon, so we'll avoid deprecation here and keep it private
-        while tries_left and bytes_read < actual_part_size:
+        start_time = time()
+        while time() - start_time < retry_time and bytes_read < actual_part_size:
             cloud_range = starting_cloud_range.subrange(
                 bytes_read, actual_part_size - 1
             )  # first attempt was for the whole file, but retries are bound correctly
             logger.debug(
-                'download attempts remaining: %i, bytes read already: %i. Getting range %s now.',
-                tries_left, bytes_read, cloud_range
+                'download attempts remaining time: %is, bytes read already: %i. Getting range %s now.',
+                int(retry_time - (time() - start_time)), bytes_read, cloud_range
```

Comment:
if an operation longer than 5min (with default settings) such as downloading a long stream on a device with poor network would fail once, it won't be retried. I think by default we should still perform some constant number of retries if the retry time is not specified.